doc/generate_config_rst.py

```diff
@@ -84,13 +86,20 @@ def print_object(output, obj, depth=0, required_entries=None, additional_desc=No
                 print_object(
                     output, v["properties"], depth=depth + 1, required_entries=reqs
                 )
+                # Strict schema with no extra fields allowed https://github.com/microsoft/CCF/issues/3813
+                assert (
+                    "allOf" in v
+                    or "additionalProperties" in v
+                    and v["additionalProperties"] is False
```

Comment:
```suggestion
                    v.get("additionalProperties") == False
```

is terser, and does not rely on precedence.